File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = INT_MIN, buyDay = prices[0];
        for(int i = 0; i < prices.size(); i++) {
            buyDay = min(buyDay, prices[i]);
            res = max(res, (prices[i] - buyDay));
        }
        return res;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = Integer.MIN_VALUE;
        int buyDay = prices[0];
        for(int i = 0; i < prices.length; i++) {
            buyDay = Math.min(buyDay, prices[i]);
            res = Math.max(res, prices[i] - buyDay);
        }
        return res;
    }
}